src/atomicassets.cpp

```diff
@@ -666,6 +640,85 @@ ACTION atomicassets::setassetdata(
     });
 }
 
+/**
+*  Updates the mutable data of a template within the templatedata table
+*  If the row doesn't exist within the template, it emplaces a new row
+*  If the new_mutable_data is empty & the row exists, it eraes the row
+*  @required_auth authorized_editor, who is within the authorized_accounts list of the collection
+                  specified in the related template
+*/
+
+
+ACTION atomicassets::settempldata(
+    name authorized_editor,
+    name collection_name,
+    name schema_name,
+    int32_t template_id,
+    ATTRIBUTE_MAP new_mutable_data
+) {
+
+    require_auth(authorized_editor);
+
+    auto collection_itr = collections.require_find(collection_name.value,
+        "No collection with this name exists");
+
+    check_has_collection_auth(
+        authorized_editor,
+        collection_name,
+        "The editor is not authorized within the collection"
+    );
+
+    schemas_t collection_schemas = get_schemas(collection_name);
+    auto schema_itr = collection_schemas.require_find(schema_name.value,
+        "No schema with this name exists");
+
+    templates_t collection_templates = get_templates(collection_name);
+    auto template_itr = collection_templates.require_find(template_id,
+        "No template with the specified id exists for the specified collection");
+
+    check_name_length(new_mutable_data);
+
+    template_data_t template_data = get_template_data(collection_name);
+    auto template_data_itr = template_data.find(template_id);
+
+    ATTRIBUTE_MAP deserialized_old_data;
+    
+    if (template_data_itr != template_data.end()){
+        deserialized_old_data = deserialize(
+            template_data_itr->mutable_serialized_data,
+            schema_itr->format
+        );
+    }
+
+    action(
+        permission_level{get_self(), name("active")},
+        get_self(),
+        name("logsetdatatl"),
+        make_tuple(collection_name, schema_name, template_id, deserialized_old_data, new_mutable_data)
+    ).send();
+
+    // If entry doesn't exist, then emplace entry
```

Comment:
Added a condition for emplacing. It's a rather unique scenario & would likely only be executed by accident by users, but this just makes it a bit cleaner internally & more consistent with the entry modification functionality, while still not throwing an error to the user & breaking the transaction.

I'm leaning towards not having a check here to avoid breaking transactions when it's not critical / absolutely necessary.

```C++
    if (template_data_itr == template_data.end() && new_mutable_data.size() > 0){
        template_data.emplace(authorized_editor, [&](auto &_template_data) {
            _template_data.template_id = template_id;
            _template_data.schema_name = template_itr->schema_name;
            _template_data.mutable_serialized_data = serialize(new_mutable_data, schema_itr->format);
        });
    }
```